src/client/v2/algod/accountApplicationInformation.ts

```diff
@@ -1,8 +1,9 @@
 import JSONRequest from '../jsonrequest';
 import HTTPClient from '../../client';
 import IntDecoding from '../../../types/intDecoding';
+import { AccountApplicationResponse } from './models/types';
 
-export default class AccountApplicationInformation extends JSONRequest {
+export default class AccountApplicationInformation extends JSONRequest<AccountApplicationResponse> {
```

Comment:
These changes only affect the typescript types that the `do` method says it returns -- no actual code change has happened.

To properly implement the change, you need to inherit from `JSONRequest<Model, Record<string, any>>` **and** override the `prepare(Record<string, any>): Model` method.